api/src/wfl/service/datarepo.clj

```diff
@@ -117,3 +117,114 @@
+;; the TDR Swagger page in order to make the
+;; request work.
+(defn create-snapshot
+  "Create a snapshot from standard SQL query,
+   assert or row-ids, based on `snapshot-request`.
+
+   See `SnapshotRequestModel` in the
+   DataRepo swagger page for more information.
+   https://jade.datarepo-dev.broadinstitute.org/swagger-ui.html#/"
+  [snapshot-request]
+  (-> (repository "snapshots")
+      (http/post {:headers      (auth/get-service-account-header)
+                  :content-type :application/json
+                  :form-params  snapshot-request})
+      util/response-body-json
+      :id
+      poll-job
+      :id))
+
+(defn list-snapshots
+  "List all Snapshots the caller has access to.
+   Hard-coded to return 999 pages for now.
+
+   Parameters
+   ----------
+   dataset-ids      - Optionally filter the result snapshots
+                      where provided datasets are source datasets.
+
+   Example
+   -------
+     (list-snapshots)
+     (list-snapshots \"48a51f71-6bab-483d-a270-3f9ebfb241cd\" \"85efdfea-52fb-4698-bee6-eef76104a7f4\")"
+  [& dataset-ids]
+  (letfn [(maybe [m k v] (if (seq v) (assoc m k {:datasetIds v}) m))]
+    (-> (http/get (repository "snapshots")
+                  (maybe {:headers (auth/get-service-account-header)
+                          :query-params {:limit 999}}
+                         :query-params dataset-ids))
+        util/response-body-json)))
+
+(defn delete-snapshot
+  "Delete the Snapshot with `snapshot-id`."
+  [snapshot-id]
+  (-> (repository "snapshots/" snapshot-id)
+      (http/delete {:headers (auth/get-service-account-header)})
+      util/response-body-json
+      :id
+      poll-job))
+
+(defn snapshot
+  "Query the DataRepo for the Snapshot with `snapshot-id`."
+  [snapshot-id]
+  (-> (repository "snapshots/" snapshot-id)
+      (http/get {:headers (auth/get-service-account-header)})
+      util/response-body-json))
+
+;; visible for testing
+(defn compose-snapshot-query
+  "Helper function for composing row-id query payload from `dataset` and `table`,
+   given a date range specified by exclusive `start` and inclusive `end`.
+
+   Note if there are no matching rows between (start, end], TDR will throw
+   a 400 exception.
+
+   Note TDR prefixes datasets in BigQuery with `datarepo_`.
+
+   Parameters
+   ----------
+   _dataset   - Dataset information response from TDR.
+   table      - Name of the table in the dataset schema to query from.
+   start      - The start date object in the timeframe to query exclusively.
+   end        - The end date object in the timeframe to query inclusively."
+  [{:keys [name dataProject] :as _dataset} table start end]
+  (let [dataset-name (str "datarepo_" name)
+        query "SELECT
+                 datarepo_row_id
+               FROM
+                 `%s.%s.%s`
+               WHERE
+                 datarepo_ingest_date > '%s' AND datarepo_ingest_date <= '%s'"]
+    (->> [dataProject dataset-name table start end] (apply format query))))
```

Comment:
I sometimes favor `compose` or `forge` over `make` and `create`, idk why though 😅 Will update to `make` if it conveys better.

Reply:
The `(]` usage was not a standard but based on my decision that when you query, you probably want "Everything added since last date I checked but also include what's added today". Not sure if makes sense to you though.

Reply:
done